Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc

```diff
@@ -107,113 +85,122 @@ void RawPCCProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup){
 //--------------------------------------------------------------------------------------------------
 void RawPCCProducer::beginLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
 
-    outputLumiInfo = std::make_unique<LumiInfo>(); 
 
     if(saveCSVFile_){
-        csvfile.open(csvOutLabel_, std::ios_base::app);
-        csvfile<<std::to_string(lumiSeg.run())<<",";
-        csvfile<<std::to_string(lumiSeg.luminosityBlock())<<",";
+        csvfile_.open(csvOutLabel_, std::ios_base::app);
+        csvfile_<<std::to_string(lumiSeg.run())<<",";
+        csvfile_<<std::to_string(lumiSeg.luminosityBlock())<<",";
     }
 }
 
 //--------------------------------------------------------------------------------------------------
 void RawPCCProducer::endLuminosityBlock(edm::LuminosityBlock const& lumiSeg, const edm::EventSetup& iSetup){
-    totalLumi_=0.0;
-    statErrOnLumi_=0.0;
+}
 
-    std::fill(clustersPerBXOutput_.begin(),  clustersPerBXOutput_.end(),0);
```

Comment:
These fills are no longer necessary since I create new containers each `endLuminosityBlockProduce` call and they are constructed with the correct size and filled with 0.